Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -114,3 +208,242 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
+/// highlighted spans of tokens.
+///
+/// The output of `preProcessFragment` is suitable for differencing, but causes unexpected results
+/// when handed directly to Swift-DocC-Render and applied to its declaration formatter. This
+/// function recombines adjacent tokens so that the declaration formatter can correctly see the
+/// structure of a declaration and format it accordingly.
+///
+/// In addition, this function takes the opportunity to trim whitespace at the beginning and end of
+/// highlighted spans of tokens, to beautify the resulting declaration.
+fileprivate func postProcessTokens(
+    _ tokens: [DeclarationRenderSection.Token]
+) -> [DeclarationRenderSection.Token] {
+    var processedTokens: [DeclarationRenderSection.Token] = []
+
+    // This function iterates a list of tokens, keeping track of a "previous token" and comparing it
+    // against the "current token" to perform the following transformations:
+    //
+    // 1. Trim whitespace from the end of a highlighted token if the following token is not
+    //    highlighted.
+    // 2. Trim whitespace from the beginning of a highlighted token if the previous token was not
+    //    highlighted.
+    // 3. Concatenate plain-text tokens if they are both highlighted or both not highlighted.
+
+    /// If the given tokens are both plain-text tokens with the same highlight, return a token with
+    /// both tokens' texts.
+    func concatenateTokens(
+        _ lhs: DeclarationRenderSection.Token,
+        _ rhs: DeclarationRenderSection.Token
+    ) -> DeclarationRenderSection.Token? {
+        guard lhs.kind == .text, rhs.kind == .text, lhs.highlight == rhs.highlight else {
+            return nil
+        }
+        return .init(cloning: lhs, withText: lhs.text + rhs.text)
+    }
+
+    var previousToken: DeclarationRenderSection.Token? = nil
+    for var currentToken in tokens {
+        if var previousToken = previousToken {
+            // First, check whether the tokens we have start or end a highlighted span, and whether
+            // that span started or ended (respectively) with whitespace. If so, break off that
+            // whitespace into a new, unhighlighted token, and save any excess tokens accordingly.
+            // This is complicated by the fact that the whitespace token could be all whitespace,
+            // removing the need to create a new token. Both of these branches also "fall through"
+            // to the concatenation check below, in case the creation of a new unhighlighted token
+            // allows it to be combined with the current token.
+
+            // If the previous token was a highlighted plain-text token that ended with whitespace,
+            // and the current token is not highlighted, then remove the highlighting for the
+            // whitespace.
+            if previousToken.isHighlightedText,
+                currentToken.highlight == nil,
+                previousToken.text.last?.isWhitespace == true
```

Comment:
Might be nice to add a function `isHighlighted` so you could write:

```suggestion
            if previousToken.isHighlightedText,
                !currentToken.isHighlighted,
                previousToken.text.last?.isWhitespace == true
```

...reads a bit better.